### DISCUSSION TITLE
Translating Javacsript and Knockout with js-translation.json

### DISCUSSION THREAD

**Discussion by CptCharlesG** [breezefront/community#226] [1. Help] [answered]
Hello,

I am trying to figure out a way to use the translation function with the language files.

As I see js-translation.json is generated, but not loaded.

In the documentation $t function does exists, and there is an example to add phrases but does not mention i18n source.
https://breezefront.com/docs/globals#translate

Is there any easy way to make this function work the same away as in Magento core?

Or should we make a function that iterates trough js-translation.json and adds the phrases?

Thank you in advance!

**Comment by vovayatsyuk** [comment, marked as answer]
It just works when using `dictionary` for `dev/js/translate_strategy`. You don't need to do anything except add the translation to the csv file:

1. Create store view with hu_HU language.
2. Create hu_HU.csv:

    ```csv
    "Next Image","Translation"
    ```

4. Add `$t('Next Image')` to your js file
5. Clean cache and static content
6. Translations will be added before closing body: `<script data-breeze>var translations = {...}`

If you need to debug something, the code is here: https://github.com/breezefront/module-breeze/blob/breeze-dynamic-js/Block/JsTranslations.php

Didn't test with `embedded` strategy, but I think it will fork fine too.

**Comment by CptCharlesG** [comment]
Thank you for your detailed help!

Originally it was set to `embedded` but now I changed `dictionary` and it works!
